lib/FieldType/DataTransformer/FormattedTimeToTimestampTransformer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+namespace EzSystems\RepositoryForms\FieldType\DataTransformer;
+
+use DateTime;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Converts between H:i(:s) format and timestamp.
+ */
+class FormattedTimeToTimestampTransformer implements DataTransformerInterface
+{
+    /** @var bool */
+    protected $withSeconds;
+
+    /**
+     * @param bool $withSeconds
+     */
+    public function __construct(bool $withSeconds)
+    {
+        $this->withSeconds = $withSeconds;
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return int|null
+     *
+     * @throws TransformationFailedException
+     */
+    public function transform($value)
+    {
+        if (empty($value)) {
+            return null;
+        }
+
+        if (!is_string($value)) {
+            throw new TransformationFailedException('Expected a string.');
+        }
+
+        $dateTime = DateTime::createFromFormat($this->withSeconds ? 'H:i:s' : 'H:i', $value);
```

Comment:
Is it possible that the function `createFromFormat()` will return bool in this case?

Reply:
I don't think so? Why would it return `true`? It should return a proper object or `false` (gotta love PHP for that inconsistency) if the `$value` cannot be parsed using provided datetime format.

Reply:
If it will return `false` for some reason, then next line will fail

Reply:
This line will fail if you provide timestamp in `$value`.